openingd/openingd.c

```diff
@@ -442,8 +442,8 @@ static u8 *funder_channel(struct state *state,
 	struct pubkey *changekey;
 	struct bitcoin_signature sig;
 	u32 minimum_depth;
-	const u8 *wscript;
-	struct bitcoin_tx *funding;
+	struct bitcoin_tx *funding = 0;
+	const u8 *wscript = 0;
```

Comment:
We generally don't use dummy initializations since that disables `valgrind`'s check of uninitialized fields. It's better to have a separate `goto` fail target that we jump to after these fields are initialized so that the `tal_free` below is only invoked once the are initialized.

Reply:
> Doesn't the default value of 1000sat trigger this failure anyway?
Not sure if I understand this question...

I will fix the dummy initialization with the next force push